Dockerfile

```diff
@@ -1,12 +1,29 @@
-FROM gcr.io/google.com/cloudsdktool/cloud-sdk:slim
-
-RUN apt-get update && \
-	apt-get -y install unzip kubectl \
-	&& \
-	git clone https://github.com/tfutils/tfenv.git ~/.tfenv && \
-	ln -s /root/.tfenv/bin/* /usr/local/bin \
-	&& \
-	tfenv install 1.3.7 && tfenv install 1.7.3
-
-ENV TFENV_AUTO_INSTALL=false
-ENTRYPOINT ["/bin/bash"]
\ No newline at end of file
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:alpine AS builder_cloud_sdk
+
+FROM alpine:3.19 AS builder_tfenv
+
+# Install tfenv
+RUN apk add --no-cache bash git && \
+    git clone --depth=1 https://github.com/tfutils/tfenv.git /tfenv
+
+FROM alpine:3.19
+# Add gcloud and tfenv to the path
+ENV PATH /google-cloud-sdk/bin:$PATH
+ENV PATH /tfenv/bin:$PATH
+
+# Install dependencies
+RUN apk add --no-cache python3 bash jq curl git
```

Comment:
Where is `curl` and `git` needed?.

Reply:
`git` is needed when we terraform load balancer in deploy infrastructure it pulls some library. `curl` I can double check but i think it's there for similar reasons.

Reply:
`curl` needed for when we install terraform version using `tfenv` at the bottom of this dockerfile, otherwise: `environment: line 9: curl: command not found`.

Reply:
<img width="800" alt="Screenshot 2024-03-26 at 10 11 39" src="https://github.com/ONSdigital/eq-infrastructure-deploy-image/assets/53475968/eddc4292-4f99-4eb8-b07f-e89742c9f1b7">